### ISSUE TITLE
[REQ/Questions] How get Magazines each month when is out?

### ISSUE THREAD

**Issue by maxcore** [DobyTang/LazyLibrarian#175]
Why LazyLibrarian don't take each month issues ?

ex: AppleMagazine October 2015 has grab but the november  issues is out but Lazy don't search releases like sickrage .

This option is implanted ?

**Comment by philborman** [comment]
It wasn't, but it is now ;-)
Just need to test it a bit more, will upload tonight or tomorrow

**Comment by maxcore** [comment]
and i like test your stuff haha ;)

**Comment by philborman** [comment]
test version in my git now
magazine rescans at same time as books, default every 360 minutes, 
changes need a restart
set in config settings, Search and Processing
new button "Show Job Status"  prints time of next run in logfile

Phil

**Comment by maxcore** [comment]
great work !

**Comment by maxcore** [comment]
@philborman 
Questions , i dont know why but the cron job check only 4 magazines above 30 ? its normal ? or i need to  readd all magazines for the new config?

![capture d ecran 2015-11-15 13 30 28](https://cloud.githubusercontent.com/assets/4043787/11170280/4f5d3770-8b9d-11e5-8fef-b18cf662a6ae.png)

the manage is for only book ?

**Comment by philborman** [comment]
yes, manage is only for books, nothing similar written for magazines yet.

I think magazines need a complete rethink, needs to behave much more 
like books.
Author -> magazine title
Book -> magazine issue

It is a lot of work though!


Cron job should check every magazine marked as "Active" in the 
lazylibrarian database.
Can you check the database contents (do you have an sqlite editor for mac?)
if not I can send you a little python script to check the database

**Comment by maxcore** [comment]
no i dont have

**Comment by maxcore** [comment]
![capture d ecran 2015-11-15 22 48 20](https://cloud.githubusercontent.com/assets/4043787/11173970/86ef058a-8beb-11e5-97e5-617eef616635.png)

strange my lazy take only the 3 first magazines in my db

**Comment by philborman** [comment]
A few reasons here, I think

1. Single-word magazine titles bug fixed in latest release in my git,
     should fix some of the missing ones.

2. LazyLibrarian only downloads issues less than one month old

3. Rejects torrents with not many seeders

Some of the magazines might fail because of that. Magazines that fail on 
numbers 2,3 are added to the "Past Issues" list, can you check if there 
are any there?  If they are, you can select for download manually.

**Comment by maxcore** [comment]
i have read each mag and now thats fixed! thx !

**Comment by philborman** [comment]
Can you post a debug log for everything after the "searching for 28 magazines"  messgage. Thanks

**Comment by theguardian** [comment]
Just wanted to chime in here since I was the original dev for the magazine feature. Magazines are difficult and can't be treated the same as other media because there is no "master database" website of magazines that the system can pull from / check against for latest issues. There's no comparable service for magazines like goodreads (authors & books), thetvdb (series & episodes), imdb (movies), musicbrainz (artists & albums).

The code is currently written as a combination of parsing filenames and checking release dates, neither of which are an easy solution. If a magazine is weekly, there's usually a date in the filename, if monthly, there's usually an abbreviated month name, if quarterly, there's usually a season name, etc - but not always.

Long story short, with no master database to pull from and no standard naming convention of releases, this feature will probably always be hit or miss. It could definitely be improved upon, though.

**Comment by maxcore** [comment]
@theguardian good point ! @philborman has do a big step for the magazine and work great ! just with the filename and month the releases was take 85% of the time !

**Comment by philborman** [comment]
theguardian is right about the naming issues,  i think there's never going
to be a perfect solution because we have no control over how the uploader
names things,  but we do pretty well at the moment.   Probably still more
tweaks needed to date recognition as we come across ones that fail. Is
there an easy way to get foreign season names?  We now do foreign month
names by switching locales on startup,  but locales dont include season
names.

The lack of a centralised database isn't so much of a problem for
magazines,  just look every x hours or days to see if there is an issue
newer than the last one you've got. As long as you scan at least once a day
you shouldnt miss anything,  even newspapers.

I think maybe we should store the magazine frequency somewhere,  not so we
can look for new issues,  but so we can do the equivalent of a library scan
to rebuild the database?

**Comment by maxcore** [comment]
'I think maybe we should store the magazine frequency somewhere,  not so we
can look for new issues,  but so we can do the equivalent of a library scan
to rebuild the database'

i think is a good idea .

you are on the good way with the last update with the cron
